node/_fs/_fs_read_test.ts

```diff
@@ -0,0 +1,143 @@
+        resolve({ fd, bytesRead, data });
+      });
+    });
+  })
+    .then(({ fd, bytesRead, data }) => {
+      fd1 = fd;
+      expected(fd, bytesRead, data);
+    })
+    .finally(() => closeSync(fd1));
+}
+
+Deno.test({
+  name: "readSuccess",
+  async fn() {
+    const moduleDir = path.dirname(path.fromFileUrl(import.meta.url));
+    const testData = path.resolve(moduleDir, "testdata", "hello.txt");
+    const buf = Buffer.alloc(1024);
+    await readTest(
+      testData,
+      buf,
+      buf.byteOffset,
+      buf.byteLength,
+      null,
+      (_fd, bytesRead, data) => {
+        assertStrictEquals(bytesRead, 11);
+        assertMatch(data.toString(), /hello world/);
+      },
+    );
+  },
+});
+
+Deno.test({
+  name:
+    "[std/node/fs] Read only five bytes, so that the position moves to five",
+  async fn() {
+    const moduleDir = path.dirname(path.fromFileUrl(import.meta.url));
+    const testData = path.resolve(moduleDir, "testdata", "hello.txt");
+    const buf = Buffer.alloc(5);
+    await readTest(
+      testData,
+      buf,
+      buf.byteOffset,
+      5,
+      null,
+      (_fd, bytesRead, data) => {
+        assertStrictEquals(bytesRead, 5);
+        assertEquals(data.toString(), "hello");
+      },
+    );
+  },
+});
+
+Deno.test({
+  name: "[std/node/fs] Specifies where to begin reading from in the file",
+  async fn() {
+    const moduleDir = path.dirname(path.fromFileUrl(import.meta.url));
+    const testData = path.resolve(moduleDir, "testdata", "hello.txt");
+    const buf = Buffer.alloc(11);
+    await readTest(
+      testData,
+      buf,
+      buf.byteOffset,
+      buf.byteLength,
+      6,
+      (_fd, bytesRead, data) => {
+        assertStrictEquals(bytesRead, 5);
+        assertEquals(
+          data,
+          Buffer.from([119, 111, 114, 108, 100, 0, 0, 0, 0, 0, 0]),
+        );
+      },
+    );
+  },
+});
+
+Deno.test({
+  name: "[std/node/fs] Read fs.read(fd, options, cb) signature",
+  async fn() {
+    const file = Deno.makeTempFileSync();
+    Deno.writeTextFileSync(file, "hi there");
+    const fd = openSync(file, "r+");
+    const buf = Buffer.alloc(11);
+    await read(
+      fd,
+      {
+        buffer: buf,
+        offset: buf.byteOffset,
+        length: buf.byteLength,
+        position: null,
+      },
+      (err, bytesRead, data) => {
+        assertEquals(err, null);
+        assertStrictEquals(bytesRead, 8);
+        assertEquals(
+          data,
+          Buffer.from([104, 105, 32, 116, 104, 101, 114, 101, 0, 0, 0]),
+        );
+      },
+    );
+    closeSync(fd);
+  },
+});
+
+Deno.test({
+  name: "[std/node/fs] Read fs.read(fd, cb) signature",
+  async fn() {
+    const file = Deno.makeTempFileSync();
+    Deno.writeTextFileSync(file, "hi deno");
+    const fd = openSync(file, "r+");
+    await read(fd, (err, bytesRead, data) => {
+      assertEquals(err, null);
+      assertStrictEquals(bytesRead, 7);
+      assertStrictEquals(data?.byteLength, 16384);
+    });
```

Comment:
`read` doesn't return promise. So awaiting it doesn't make much sense. Because we want to confirm that the callback is actuall called, I think the these lines should be something like:
```js
await new Promise((resolve) => {
  read(fd, (err, bytesRead, data) => {
    assertEquals(err, null);
    assertStrictEquals(bytesRead, 7);
    assertStrictEquals(data?.byteLength, 16384);
    resolve();
  });
});
```